### PR TITLE
Bsweger/retrieve complex forecast data from s3

### LINKED ISSUE #20: vignette describing forecast data

### PR DESCRIPTION
**Pull request by bsweger**
Resolves #20

Now that we've changed org names and have properly onboarded example-complex-forcast-hub to S3, this PR switches the corresponding `data-raw/` script to use S3 as its data source (versus a local, cloned version of example-complex-forecast-hub).

Not sure if this is a "proper" practice, but to ensure that these changes don't alter the contents of the .rda files, there's also a new test suite that can be run locally. These tests don't run as part of R checks because the contents of `data-raw/` aren't part of the `hubExamples` package.

Output of `devtools::test()`:

```
══ Results ════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════
Duration: 6.2 s

[ FAIL 0 | WARN 0 | SKIP 0 | PASS 3 ]
```

### CODE REVIEW

File: tests/testthat/testdata/forecast_outputs.rda
Comment (bsweger) [inline tests/testthat/testdata/forecast_outputs.rda:1]: The `.rda` files in `testdata` are copies of their counterparts in `data/` (from the main branch, so we can compare the data before and after the S3 changes).

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -30,15 +30,18 @@ Our procedures for contributing bigger changes, code in particular, generally fo
 
 *  For user-facing changes, add a bullet to the top of `NEWS.md` (i.e. just below the first header). Follow the style described in <https://style.tidyverse.org/news.html>.
 
+*  If you're changing the scripts in `data-raw/`, you can run tests locally: `devtools::test()` (because the
```
Comment (bsweger) [inline .github/CONTRIBUTING.md:33]: This is the only real change to `CONTRIBUTING.md`. The rest of the diff is a pre-commit check the removes trailing whitespace.

File: tests/testthat/helper-path.R [hunk 2]
```diff
@@ -0,0 +1,2 @@
+code_path <- dirname(dirname(getwd()))
```
Comment (bsweger) [inline tests/testthat/helper-path.R:1]: This is so janky. It's as if `testthat` doesn't want anyone to invoke non-package files 😬 

(these tests did help me, though!)

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -30,15 +30,18 @@ Our procedures for contributing bigger changes, code in particular, generally fo
...
+contents of `data-raw` are not included in the package, these tests are skipped during R checks).d
```
Comment (elray1) [inline .github/CONTRIBUTING.md:34]:
```suggestion
contents of `data-raw` are not included in the package, these tests are skipped during R checks).
```

File: tests/testthat/test-generate_example_forecast_data.R [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+# These tests are designed to help devs check their work when updating the logic that generates hubExample's
+# data files. Because these scripts (i.e., the contents of data-raw/) are not included in the hubExample
+# package itself, the tests will not run during R checks.
+skip_if(is_checking())
+
+# Source the script that generates the example forecast data so we can test the functions
+source(file.path(forecast_data_path, "generate_example_forecast_data.R"))
+
+
+test_that("forecast_outputs dataset is generated correctly", {
```
Comment (elray1) [inline tests/testthat/test-generate_example_forecast_data.R:12]: i can sense your burning anguish at the lack of parameterized tests here.

File: tests/testthat/helper-path.R (hunk 2 above)
Comment (elray1) [reply tests/testthat/helper-path.R:1]: this is not great!  i probably found the same stackoverflow post you did, indicating there is not a better answer.

**Review by elray1 — APPROVED** [review]
approved pending minor typo fix, all else looks good (given the constraints of the testing system)

File: tests/testthat/test-generate_example_forecast_data.R (hunk 4 above)
Comment (bsweger) [reply tests/testthat/test-generate_example_forecast_data.R:12]: It's true--testing in general is not the best in R imo (though snaps are pretty great!)

File: tests/testthat/helper-path.R (hunk 2 above)
Comment (zkamvar) [reply tests/testthat/helper-path.R:1]: This makes my skin crawl. 

Wouldn't a better solution for this be to include a line in the `data-raw/` script that copies itself to `tests/testthat/testdata`?

**Pull request merged**